services/movie.py

```diff
@@ -0,0 +1,45 @@
+from django.db.models import QuerySet
+
+from db.models import Movie
+
+
+def get_movies(genres_ids: list[int] = None,
+               actors_ids: list[int] = None) -> QuerySet:
+    movies = Movie.objects.all()
+
+    if not genres_ids and not actors_ids:
+        return movies
+
+    if genres_ids and actors_ids:
+        movies = movies.filter(genres__id__in=genres_ids)
+        movies = movies.filter(actors__id__in=actors_ids)
+        return movies
+
+    if genres_ids:
+        movies = movies.filter(genres__id__in=genres_ids)
+        return movies
+
+    if actors_ids:
+        movies = movies.filter(actors__id__in=actors_ids)
+        return movies
+
+
+def get_movie_by_id(movie_id: int) -> Movie:
+    if Movie.objects.filter(id=movie_id).exists():
+        return Movie.objects.get(id=movie_id)
```

Comment:
Using `filter().exists()` followed by `get()` can lead to race conditions. It's better to use `get_object_or_404` or handle the `DoesNotExist` exception to ensure the application fails gracefully and provides user-friendly error messages.